### PR TITLE
Configure if CI is required or not

### PR DESCRIPTION
**Pull request by nilsgstrabo**
Added env var REQUIRE_APP_CONFIGURATION_ITEM to control if Configuration Item is required when registering/updating application

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -35,11 +35,15 @@ import (
 	_ "github.com/equinor/radix-api/docs"
 )
 
-const clusternameEnvironmentVariable = "RADIX_CLUSTERNAME"
+const (
+	clusternameEnvironmentVariable = "RADIX_CLUSTERNAME"
```
Comment (satr) [inline main.go:39]: Maybe take "RADIX_CLUSTERNAME" name from radix-operator defaults?

File: main.go [hunk 2]
```diff
@@ -35,11 +35,15 @@ import (
...
+	// requireAppConfigrationItemEnvironmentVariable = "REQUIRE_APP_CONFIGURATION_ITEM"
```
Comment (satr) [inline main.go:40]: This comment does not much explain. Also has a typo "Config*u*ration"

File: api/applications/applications_handler_config.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package applications
+
+import (
+	"github.com/equinor/radix-api/internal/flags"
+	"github.com/mitchellh/mapstructure"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+type ApplicationHandlerConfig interface {
```
Comment (satr) [inline api/applications/applications_handler_config.go:10]: Maybe call it `ApplicationConfig` - not specifically for handler?

File: main.go [hunk 4]
```diff
@@ -95,7 +108,15 @@ func getControllers() []models.Controller {
...
 		secrets.NewSecretController(),
+	}, nil
+}
+
+func getApplicationHandlerFactory() (applications.ApplicationHandlerFactory, error) {
+	cfg, err := applications.LoadApplicationHandlerConfig(os.Args[1:])
+	if err != nil {
+		return nil, err
 	}
+	return applications.NewApplicationHandlerFactory(cfg), nil
```
Comment (satr) [inline main.go:119]: Extract config to own var, which can be passed to any handler?

**Review by satr — CHANGES_REQUESTED** [review]
Suggested changes

File: main.go (hunk 1 above)
Comment (nilsgstrabo) [reply main.go:39]: fixed

File: main.go (hunk 2 above)
Comment (nilsgstrabo) [reply main.go:40]: Removed.

**Review by satr — APPROVED** [review]

**Pull request merged**